packages/runtime/runtime-utils/src/dataStoreHelpers.ts

```diff
@@ -12,45 +12,68 @@ interface IResponseException extends Error {
 	message: string;
 	code: number;
 	stack?: string;
-	underlyingResponseHeaders?: { [key: string]: any };
+	underlyingResponseHeaders?: { [key: string]: unknown };
 }
 
 /**
+ * Type guard for determining if an error is an {@link IResponseException}.
  * @internal
  */
-export function exceptionToResponse(err: any): IResponse {
+function isResponseException(err: unknown): err is IResponseException {
+	return (
+		err !== null &&
+		typeof err === "object" &&
+		"errorFromRequestFluidObject" in err &&
+		(err as { errorFromRequestFluidObject: unknown }).errorFromRequestFluidObject === true
+	);
+}
+
+/**
+ * Converts an error object into an {@link @fluidframework/core-interfaces#IResponse}.
+ * @internal
+ */
+export function exceptionToResponse(error: unknown): IResponse {
 	const status = 500;
-	if (err !== null && typeof err === "object" && err.errorFromRequestFluidObject === true) {
-		const responseErr: IResponseException = err;
+	if (isResponseException(error)) {
 		return {
 			mimeType: "text/plain",
-			status: responseErr.code,
-			value: responseErr.message,
+			status: error.code,
+			value: error.message,
 			get stack() {
-				return responseErr.stack;
+				return error.stack;
 			},
-			headers: responseErr.underlyingResponseHeaders,
+			headers: error.underlyingResponseHeaders,
 		};
 	}
 
-	// Capture error objects, not stack itself, as stack retrieval is very expensive operation, so we delay it
+	// Capture error objects, not stack itself, as stack retrieval is very expensive operation
 	const errWithStack = generateErrorWithStack();
 
 	return {
 		mimeType: "text/plain",
 		status,
-		value: `${err}`,
+		value: `${error}`,
 		get stack() {
-			return (err?.stack as string | undefined) ?? errWithStack.stack;
+			// Use type assertion after checking if error is an object with stack
+			return (
+				(typeof error === "object" && error !== null && "stack" in error
+					? (error.stack as string | undefined)
+					: undefined) ?? errWithStack.stack
+			);
 		},
 	};
 }
 
 /**
+ * Converts an {@link @fluidframework/core-interfaces#IResponse} back into an Error object that can be thrown.
+ * @param response - The {@link @fluidframework/core-interfaces#IResponse} to convert.
+ * @param request - The original {@link @fluidframework/core-interfaces#IRequest}.
+ * @returns An Error object with additional properties from the response
  * @internal
  */
 export function responseToException(response: IResponse, request: IRequest): Error {
-	const message = response.value;
+	// IResponse.value is typed as 'any', but we expect it to be a string error message
```

Comment:
I'm not sure we really _expect_ it to. I think we should rather be conservative with what we think is happening in code we don't own.
```suggestion
	// As of 2025-08-20 the code seems to assume `response.value` is always a string.
	// This type assertion just encodes that assumption as we move to stricter linting rules, but it might need to be revisited.
```